bin/tx-prover/src/utils.rs

```diff
@@ -1,20 +1,71 @@
 use std::time::Duration;
 
+use opentelemetry::{trace::TracerProvider as _, KeyValue};
+use opentelemetry_sdk::{
+    runtime,
+    trace::{RandomIdGenerator, Sampler, TracerProvider},
+    Resource,
+};
+use opentelemetry_semantic_conventions::{
+    resource::{SERVICE_NAME, SERVICE_VERSION},
+    SCHEMA_URL,
+};
 use pingora::{http::ResponseHeader, Error, ErrorType};
 use pingora_proxy::Session;
 use tonic::transport::Channel;
 use tonic_health::pb::health_client::HealthClient;
+use tracing::Level;
+use tracing_subscriber::{layer::SubscriberExt, Registry};
+
+pub const TRACING_TARGET_NAME: &str = "miden-tx-prover";
 
 const RESOURCE_EXHAUSTED_CODE: u16 = 8;
 
-pub(crate) fn setup_tracing() {
-    // Set a default log level if `RUST_LOG` is not set
-    if std::env::var("RUST_LOG").is_err() {
-        std::env::set_var("RUST_LOG", "info"); // Default to 'info' level
-    }
-    tracing_subscriber::fmt::init();
+// Construct TracerProvider for OpenTelemetryLayer
+pub(crate) fn init_tracer_provider() -> TracerProvider {
```

Comment:
nit: let's use `///` for doc comments (here and in other places in this file).

Also, could we add more details about how we configure the tracing provider? For example, why do we need to add ID generator, what does `with_sampler()` do etc.